src/configurator/lib/playground-configurator.ts

```diff
@@ -0,0 +1,478 @@
+      case 'select':
+        return [label, this.selectKnob(knob)];
+      case 'slider':
+        return [label, this.sliderKnob(knob)];
+      case 'color':
+        return [label, this.colorKnob(knob)];
+      case 'checkbox':
+        return [label, this.checkboxKnob(knob)];
+    }
+    return '';
+  }
+
+  private selectKnob(knob: KnobsOfType<'select'>) {
+    const value = this.values.getValue(knob.id);
+    return html`
+      <select
+        id=${knob.id}
+        @input=${(event: Event & {target: HTMLSelectElement}) => {
+          this.setValue(knob.id, event.target.value as any);
+        }}
+      >
+        ${(knob.options as ReadonlyArray<string>).map(
+          (option) =>
+            html`<option value=${option} ?selected=${option === value}>
+              ${option}
+            </option>`
+        )}
+      </select>
+    `;
+  }
+
+  private sliderKnob(knob: KnobsOfType<'slider'>) {
+    const value = this.values.getValue(knob.id);
+    return html`
+      <span class="sliderAndValue">
+        <input
+          id=${knob.id}
+          type="range"
+          min=${knob.min}
+          max=${knob.max}
+          value=${value}
+          @input=${(event: Event & {target: HTMLInputElement}) => {
+            this.setValue(knob.id, Number(event.target.value));
+          }}
+        />
+        <span class="sliderValue"
+          >${knob.formatCss ? knob.formatCss(value) : value}</span
+        >
+      </span>
+    `;
+  }
+
+  private colorKnob(knob: KnobsOfType<'color'>) {
+    let value = this.values.getValue(knob.id);
+    return html`
+      ${'unsetLabel' in knob
+        ? html`
+            <div style="margin-bottom: 5px">
+              <input
+                id=${knob.id + '-unset'}
+                type="checkbox"
+                ?checked=${value === ''}
+                @input=${(event: Event & {target: HTMLInputElement}) => {
+                  value = event.target.checked ? '' : '#ffffff';
+                  this.setValue(knob.id, value);
+                }}
+              />
+              <label for=${knob.id + '-unset'}> ${knob.unsetLabel}</label>
+            </div>
+            <span></span>
+          `
+        : ''}
+      <input
+        id=${knob.id}
+        type="color"
+        .value=${value === '' ? '#ffffff' : String(value)}
+        ?disabled=${value === ''}
+        @input=${(event: Event & {target: HTMLInputElement}) => {
+          this.setValue(knob.id, event.target.value);
+        }}
+      />
+    `;
+  }
+
+  private checkboxKnob(knob: KnobsOfType<'checkbox'>) {
+    let value = this.values.getValue(knob.id);
+    return html`
+      <input
+        id=${knob.id}
+        type="checkbox"
+        ?checked=${value}
+        @change=${(event: Event & {target: HTMLInputElement}) => {
+          this.setValue(knob.id, event.target.checked);
+        }}
+      />
+    `;
+  }
+
+  /**
+   * https://tholman.com/github-corners/
+   */
+  private get githubCorner() {
+    return html`<a
```

Comment:
This doesn't need to be a method, and it might be nice to put it in another file... or could stay just like this.

Reply:
Moved to a const.